_posts/2017-03-19-server-side-caracteristicas-de-um-servidor.markdown

```diff
@@ -0,0 +1,394 @@
+servidor de software de maneira a usar as principais características do sistema
+operacional no qual estará hospedado.
+
+Hoje em dia o papel principal do servidor é prover serviços a clientes leves, ou
+_thin clients_, normalmente aplicativos para celulares. Todo o processamento
+pesado e o armazenamento fica por conta do servidor, deixando o cliente com a
+tarefa de organizar, logicamente, os fluxos de trabalho disponíveis para que o
+usuáio seja atendido em determinada funcionalidade.
+
+O servidor pode ser uma ou mais peças de software trabalhando em conjunto,
+formando um sistema que pode ser agregado ou distribuído. Para o cliente, o
+servidor é um só, mesmo que sejam várias máquinas físicas diferentes, vários
+sistemas de software criados por várias linguagens diferentes, interagindo entre
+si através das mais diversas formas de integração.
+
+# Linux e Unix
+
+Nos dias de hoje o sistema operacional mais popular para _back-end_ é, sem
+dúvida, o linux. O unix é um sistema muito popular no ambiente corporativo,
+apesar das vendas deste sistema decrescer ano após ano. Sendo rigoroso, o unix
+não é linux e linux não é unix. São sistemas completamente diferentes, apesar
+das suas similaridades.
+
+Apesar do foco deste artigo ser o Linux, os princípios e conceitos aqui
+apresentados valem para o unix. Vale lembrar que o FreeBSD é um unix e não
+linux. A camada de compatibilidade entre o BSD e o linux é tão boa e
+transparente que ela faz com que praticamente tudo o que executa em um sistema
+seja executado em outro.
+
+De toda forma vou intercambiar os termos unix e linux ao longo do texto,
+tratando tudo como se fosse unix por questão de simplicidade. Se houver algo que
+seja inerente a um sistema, deixarei explícito para que não haja confusões.
+
+Vamos começar por alguns conceitos básicos que serão usados através do restante
+do texto. Com o intuito de equalizar o conhecimento, apresento de forma bastante
+rápida estes conceitos.
+
+## Daemon
+
+Apesar do termo lembrar os demônios cristãos, ou seja, entidades malévolas, o
+termo _daemon_ foi emprestado do termo grego _daimon_ que refere-se a entidades
+naturais benignas e que estão sempre presentes em nossas vidas.
+
+A despeito da questão religiosa, um _daemon_ é, na verdade, um processo autônomo
+que é executado em segundo plano, fornecendo ou executando algum tipo de
+serviço. Normalmente os daemons no unix/linux são processos iniciados durante a
+carga do sistema, através de um sistema de inicialização como o System V ou o
+upstart do Ubuntu.
+
+Uma vez iniciados, estes processos normalmente terminam quando o sistema é
+reiniciado, ou seja, são processos de execução longa, projetados para
+permanecerem funcionando por longos períodos de tempo.
+
+## IPC
+
+IPC é uma sigla que significa _Inter-Process Communication_. Tanto o unix quanto
+o linux implementam várias formas de comunicação inter-processo:
+
+- _pipes_ para escrita e leitura síncrona de dados;
```

Comment:
Quem é síncrona aqui é só a leitura, ou leitura e escrita? Se for leitura e escrita, o certo não seria escrita e leitura síncronas?